internal/worker/json.go

```diff
@@ -256,6 +256,16 @@ type ContainerResolveJobResult struct {
 	JobResult
 }
 
+type FileResolveJob struct {
+	URLs []string `json:"urls"`
+}
+
+type FileResolveJobResult struct {
+	Contents [][]byte `json:"contents"`
```

Comment:
Wouldn't it make sense to somehow tie the returned contents to the URLs that were passed in the job? So either making this something like:

```go
type FileResolveJobResultItem struct {
	URL      string `json:"url"`
	Content  []byte `json:"content"`
}

type FileResolveJobResult struct {
	Results []FileResolveJobResultItem `json:"results"`
 	JobResult
}
```

or

```go
type FileResolveJobResult struct {
	Contents map[string][]byte `json:"contents"`
 	JobResult
}
```

I know that the index of the content in the returned result will be probably the same as that the one passed in the job itself. I just find it nicer if the result content is useful on its own, without having the access to another data structure that was passed when requesting the job.

Just something to consider...

Reply:
Yeah actually I've been wondering about this. I think I had something similar in the first iteration but I removed it. I think this would be quite helpful to keep track of the url for use later down the road. I also agree that keeping track of this is nicer (and more explicit) than just relying on the indexes being the same (since we're using a goroutine, there's a chance that they might not actually be the same).

Reply:
I went with the first one since it seems more readable to me :)